Update styling to match elastic.co/docs

colleenmcginnis opened the pull request:
This PR updates the styles in our AsciiDoc-powered docs to match the Docsmobile styling used on docs.elastic.co. This will allow AsciiDoc- and Docsmobile-powered doc sets to exist together without impacting the reader experience.

## To do

### Development and design

- [x] Add initial layout and styles from docs.elastic.co
- [x] Refine responsive design
- [x] Closely review all layout elements:
  - [x] left nav
  - [x] main content
  - [x] on this page
  - [x] breadcrumbs
  - [x] version selector
- [x] Closely review all content UI elements:
    - [x] Added, coming, and deprecated notes (65d9d5c676f2d5bd4359d58d7d9457d51da2dd7a)
    - [x] Admonition blocks
    - [x] Beta, Dev, and Preview
    - [x] Code blocks
    - [x] Collapsible block
    - [x] Example blocks
    - [x] Headings
    - [x] Images
    - [x] Inline text
    - [x] Links
    - [x] Lists
    - [x] Paragraphs
    - [x] Sidebars
    - [x] Tabbed widgets
    - [x] Tables
    - [x] Thematic breaks
- [x] Clean up code

### Final QA/QC

#### Interaction testing

@colleenmcginnis: 

- [x] Version selector
- [x] Tab widget testing
- [x] Collapsible block testing
- [x] Console code block
- [x] Code blocks with callouts
- [x] Code block copy button

#### Cross-browser / screen size testing

- [x] @colleenmcginnis Chrome
    - [x] Large screen size
    - [x] Laptop screen size
    - [x] Tablet size
    - [x] Mobile size
- [x] @colleenmcginnis Firefox
    - [x] Large screen size
    - [x] Laptop screen size
    - [x] Tablet size
    - [x] Mobile size
- [x] @colleenmcginnis Safari
    - [x] Large screen size
    - [x] Laptop screen size
    - [x] Tablet size
    - [x] Mobile size
- [ ] TBD Edge
    - [ ] Large screen size
    - [ ] Laptop screen size
    - [ ] Tablet size
    - [ ] Mobile size

#### Cross-book testing

Visually inspect a random sample of docs from each book your team owns. In a comment in this PR, note any edge cases that don't look quite right:

- [x] @elastic/obs-docs 
- [x] @elastic/platform-docs 
- [x] @elastic/security-docs 

### Update tests

- [x] Make sure all tests pass

### Related work

- [ ] Remove custom Javascript in index pages in Observability book, Elasticsearch book, Kibana book, etc.

colleenmcginnis commented on resources/web/template.html:
```diff
@@ -68,76 +68,142 @@
+                    <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16" class="euiIcon eui-styles-2txxin-euiIcon-m" role="img" aria-hidden="true"><path d="M2 4V2h2v2H2zm5 0V2h2v2H7zm5 0V2h2v2h-2zM2 9V7h2v2H2zm5 0V7h2v2H7zm5 0V7h2v2h-2zM2 14v-2h2v2H2zm5 0v-2h2v2H7zm5 0v-2h2v2h-2z"></path></svg>
+                    <div id="mobile-site-nav-header-tooltip">
+                      <div class="euiHeaderLinks__mobileList docs-i6hqpf-euiHeaderLinks__mobileList">
+                        <a class="euiButtonEmpty headerRightSideLink docs-xla6n6-euiButtonDisplay-euiButtonEmpty-m-empty-text" href="https://www.elastic.co/guide/en/welcome-to-elastic/current/new.html" rel="noreferrer">
```
```suggestion
                        <a class="euiButtonEmpty headerRightSideLink docs-xla6n6-euiButtonDisplay-euiButtonEmpty-m-empty-text" href="https://www.elastic.co/guide/en/starting-with-the-elasticsearch-platform-and-its-solutions/current/new.html" rel="noreferrer">
```

colleenmcginnis commented:
@elasticmachine, run elasticsearch-ci/docs

KOTungseth commented:
The Kibana landing page has turned into a Picasso

![Screenshot 2024-05-15 at 11 04 32 AM](https://github.com/elastic/docs/assets/38990715/5d4a7a32-6ec0-404b-b007-b850366c107f)

kilfoyle commented:
I've checked through the Fleet and Elastic Agent Guide and everything looks great! Even weirdly formatted tables, etc. All good. 👍

benironside commented:
Found one very minor thing while spot checking the security docs. 
![side-by-side-spaces](https://github.com/elastic/docs/assets/91905639/343021eb-33e6-42b8-a8c0-83463e626493)

https://docs.elastic.co/serverless/security/connect-to-bedrock

nastasha-solomon commented:
@colleenmcginnis should images [display in full screen](https://docs.elastic.dev/docs/reference/basic-mdx#images) when you click on them? I spot-checked images throughout the Security docset and none of them are clickable or expandable.

colleenmcginnis commented:
>@colleenmcginnis should images [display in full screen](https://docs.elastic.dev/docs/reference/basic-mdx#images) when you click on them? I spot-checked images throughout the Security docset and none of them are clickable or expandable.

No. That will only be in Docsmobile.

karenzone commented:
Everything I checked for Elastic Reference Architectures, the Logstash Reference, and the Logstash Versioned Plugin Reference looks great! Thank you for bumping this project up in the priority list.  💟

bmorelli25 commented:
I clicked around the Observability Guide and a couple of the APM Agent References. Looks good 🚀

dedemorton commented:
I spot checked quite a few topics in the Observability Guide. It looks fantastic!

There's only one thing I think we should consider changing. The right-side nav looks too much like a ransom note when words are highlighting. Plus it's harder to scan the TOC:

For example: https://docs_bk_2768.docs-preview.app.elstc.co/guide/en/observability/current/logs-parse.html

![image](https://github.com/elastic/docs/assets/14206422/e96dad05-313b-4dd5-add7-281abc921d7f)

leemthompo commented:
@colleenmcginnis you've pulled another magical rabbit from your hat! 🐰 🎩   ✨

Nothing stood out from my pseudorandom sampling of pages. 🔍

I did notice one small, *subtle* difference in left nav clickability. You now have to click exactly on the right angle bracket `>` to untoggle nested nav content. In the current system you can click anywhere on the continuum between the text and the right angle bracket. The margin for error has shrunk and inaccurate clickers might lose seconds [^1] of their lives. 😅

[^1]: In worst case scenarios. On average (for those impacted) this will probably be on the order of milliseconds.

florent-leborgne commented:
Thanks @colleenmcginnis!! I did some spot checks in the ESS and ECE cloud guides, and things look great including tabs, code callouts, tables, accordions, etc. 👍

jmikell821 commented:
Click around in Security and other than seeing the same spacing issue that Ben [previously commented](https://github.com/elastic/docs/pull/2768#issuecomment-2113439422) on, it looks great! 🌟

szabosteve commented:
I checked quite some pages under Elastic Stack (in the Installation and Upgrade Guide, Clients, ML, Painless, etc.). They all look great! Thanks, Colleen! 🚢

dedemorton commented:
@colleenmcginnis What's the schedule for getting this PR finalized? I noticed today that relative fonts in the published docs are really messed up, then Brandon pointed out that this PR fixes the problem. Here's an example of where things are really messed up: https://www.elastic.co/guide/en/observability/current/apm-data-model-metrics.html.

colleenmcginnis commented:
Thanks to everyone who tested this out! I've addressed all feedback from above: 

>@KOTungseth:
>The Kibana landing page has turned into a Picasso

Fixed in [`9cff6e2` and `11a8598`](https://github.com/elastic/docs/compare/9cff6e28f6fe5e520742f7ab52f9c34b8b46f94f..11a8598d97ca20f954986b1c653f86fb2ecade80?expand=1)

>@benironside:
>Found one very minor thing while spot checking the security docs.

Fixed in https://github.com/elastic/docs/pull/2768/commits/4db9663cb2851a37e85c197d69f6e97ffdc49520

> @dedemorton 
>The right-side nav looks too much like a ransom note when words are highlighted. Plus it's harder to scan the TOC:
>
>For example: https://docs_bk_2768.docs-preview.app.elstc.co/guide/en/observability/current/logs-parse.html

Fixed in ad3c97a9c8cd1bc5e9aaeb0265bdee8b3c25c317

>@leemthompo 
>I did notice one small, subtle difference in left nav clickability. You now have to click exactly on the right angle bracket > to untoggle nested nav content. In the current system you can click anywhere on the continuum between the text and the right angle bracket. The margin for error has shrunk and inaccurate clickers might lose seconds [1](https://github.com/elastic/docs/pull/2768#user-content-fn-1-245d7c9db75b677ea6471cbba0d66daf) of their lives. 😅

Fixed in caffe049db8474b1c7af61e51582104e5fe39537

KOTungseth reviewed (approved):
This is a work of art.

colleenmcginnis commented:
To do on Thursday before this is ready to merge:

- [ ] Catch up to `master`
- [ ] Run tests locally and update them to work with latest from `master`
    * Note to self: I already have changes to tests in a stash locally to get tests to pass with the changes from #1614.
- [ ] Add the edit button to h1 headings
    * Note to self: cherry-pick from https://github.com/elastic/docs/pull/3015 and fix tests
- [ ] Bump CSS and JS files to `v2` using the methods we used in https://github.com/elastic/docs/pull/3017 and https://github.com/elastic/docs/pull/3019
- [ ] Check preview
- [ ] Tag @bmorelli25 to review

bmorelli25 reviewed (dismissed):
New additions LGTM

colleenmcginnis commented:
After chatting with @bmorelli25, we're planning to delay the deployment of these changes to July 2 in an abundance of caution (since we'll have reduced capacity to address any issues that are reported next week). I will address the missing edit links in #3015 instead so we can get that small fix out ahead of merging this PR. 

cc @KOTungseth @bmorelli25 @timto-elastic

bmorelli25 commented:
Thank you so much for all of the time and effort that you put into this PR. While there are significant look and feel improvements in this PR, there are also complexities related to making elastic.co/guide look and act like Docsmobile. Considering our new plan for Docsmobile, I don't think it makes sense to merge this PR.

dedemorton commented:
Bummer. I think for serverless users, jumping between the two doc sets is jarring. I've been looking forward to these fixes for a long time, especially since they resolve some technical issues (like topics that do not have mini TOCs). Would it be possible to pull out the changes that aren't just visual and make sure that asciidoc output is fixed?
